webauthn/helpers/structs.py

```diff
@@ -475,6 +474,52 @@ class AttestationObject(WebAuthnBaseModel):
     auth_data: AuthenticatorData
     att_stmt: AttestationStatement = AttestationStatement()
 
+class LargeBlobSupport(str, Enum):
+    """A Relying Party's requirement for large blob support.
+
+    Members:
+        `REQUIRED`: Large blob support is required.
+        `PREFERRED`: Large blob support is preferred, but operation will not fail if the authenticator does not support large blobs.
+
+    https://www.w3.org/TR/webauthn-2/#enumdef-largeblobsupport
+    """
+
+    REQUIRED = "required"
+    PREFERRED = "preferred"
+
+# See https://www.w3.org/TR/webauthn-2/#sctn-large-blob-extension
+class AuthenticationExtensionsLargeBlobInputs(WebAuthnBaseModel):
+    """A Relying Party's input for the large blob extension.
+
+    Attributes:
+        (optional) `support`: Whether large blob support is requested
+        (optional) `read`: Whether the large blob content shall be read from the authenticator
+        (optional) `write`: The data that shall be written to the authenticator
+
+    https://www.w3.org/TR/webauthn-2/#dictdef-authenticationextensionslargeblobinputs
+    """
+    support: Optional[LargeBlobSupport] = None
+    read: Optional[bool] = None
+    write: Optional[bytes] = None
+
+class AuthenticationExtensionClientInputs(WebAuthnBaseModel):
+    # See https://www.w3.org/TR/webauthn-2/#sctn-large-blob-extension
+    large_blob: Optional[AuthenticationExtensionsLargeBlobInputs] = None
+
+class CredentialProtectionPolicy(str, Enum):
+    """Various registered values indicating whether a credential shall be protected (influences how discoverable credentials are handled).
+
+    Members:
+        `USER_VERIFICATION_OPTIONAL`
+        `USER_VERIFICATION_OPTIONAL_WITH_CREDENTIAL_ID_LIST`
+        `USER_VERIFICATION_REQUIRED`
+
+    https://fidoalliance.org/specs/fido-v2.1-ps-20210615/fido-client-to-authenticator-protocol-v2.1-ps-20210615.html#sctn-credProtect-extension
+    """
+    USER_VERIFICATION_OPTIONAL = 'userVerificationOptional'
+    USER_VERIFICATION_OPTIONAL_WITH_CREDENTIAL_ID_LIST = 'userVerificationOptionalWithCredentialIDList'
+    USER_VERIFICATION_REQUIRED = 'userVerificationRequired'
+
```

Comment:
I understand the value of this extension, but my goal is for this library to support only the functionality defined in the WebAuthn spec itself. Since this extension isn't formally defined in https://www.w3.org/TR/webauthn-2/#sctn-defined-extensions then I'm inclined to request that it not be included in this diff.

Reply:
There's some conversation going on in an issue in the spec repo that's educating me on how my position in which extensions to support may not be reasonable: https://github.com/w3c/webauthn/pull/1703#discussion_r846202858

I'm going to revisit this PR next week when I get back in the office, your `credProtect` contribution will likely make it through in one way or another.